qiita_db/search.py

```diff
@@ -194,7 +200,8 @@ def __call__(self, searchstr, user):
         Metadata column names and string searches are case-sensitive
         """
         study_sql, sample_sql, meta_headers = \
-            self._parse_study_search_string(searchstr, True)
```

Comment:
Ok, broken into three blocks: parsing query into the where clause and metadata headers searched over, building the actual query, and a helper to prepend SQL table labels.